### DISCUSSION TITLE
ScriptReference/Physics2D.Raycast

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#562] [Notes]
# ScriptReference/Physics2D.Raycast

https://docs.unity3d.com/ScriptReference/Physics2D.Raycast.html

**Comment by jschiff** [comment]
There is an overload that takes a ContactFilter2D and returns a single hit which seems to be undocumented:

https://github.com/Unity-Technologies/UnityCsReference/blob/0355e09029fa1212b7f2e821f41565df8e8814c7/Modules/Physics2D/ScriptBindings/Physics2D.bindings.cs#L139

**Reply by boxfriend** [reply]
This is actually an overload for PhysicsScene2D.Raycast rather than Physics2D.Raycast
